apps/web/src/pages/Profile/AccountSettings/NotificationSettings.tsx

```diff
@@ -0,0 +1,310 @@
+import CheckIcon from "@heroicons/react/24/solid/CheckIcon";
+
+import {
+  Checkbox,
+  CheckboxOption,
+  Field,
+  Submit,
+} from "@gc-digital-talent/forms";
+import { NotificationFamily, graphql } from "@gc-digital-talent/graphql";
+import { Card } from "@gc-digital-talent/ui";
+import { commonMessages, formMessages } from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+
+import { dataValuesToFormValues, formValuesToData } from "./utils";
+import { FormValues } from "./types";
+
+export type UpdateNotificationInput = {
+  ignoredEmailNotifications?: [NotificationFamily];
+  ignoredInAppNotifications?: [NotificationFamily];
+};
+
+interface NotificationChecklistProps {
+  id: string;
+  name: string;
+  legend: string;
+  subtitle: string;
+  disabled?: boolean;
+}
+
+const NotificationChecklist = ({
+  id,
+  name,
+  legend,
+  subtitle,
+  disabled,
+}: NotificationChecklistProps) => {
+  const intl = useIntl();
+
+  const notificationOptions: CheckboxOption[] = [
+    {
+      value: "email",
+      label: (
+        <span data-h2-visually-hidden="base(invisible)">
+          {intl.formatMessage(commonMessages.email)}
+        </span>
+      ),
+    },
+    {
+      value: "inApp",
+      label: (
+        <span data-h2-visually-hidden="base(invisible)">
+          {intl.formatMessage(commonMessages.inApp)}
+        </span>
+      ),
+    },
+  ];
+
+  if (disabled) {
+    return (
+      <div
+        data-h2-display="base(grid)"
+        data-h2-grid-template-columns="base(4fr 1fr)"
+        data-h2-align-items="base(center)"
+        data-h2-margin-bottom="base(x1)"
+      >
+        <div>
+          <p
+            data-h2-font-weight="base(bold)"
+            data-h2-padding-bottom="base(x.5)"
+          >
+            {legend}
+          </p>
+          <p data-h2-font-size="base(caption)">{subtitle}</p>
+        </div>
+        <div
+          data-h2-display="base(flex)"
+          data-h2-justify-content="base(space-between)"
+        >
+          {notificationOptions.map(({ value }) => {
+            const key = `${id}-${value}`;
+            return (
+              <span
+                key={key}
+                data-h2-padding="base(x.25 x.75 x.25 x.5)"
+                data-h2-color="base(gray)"
+              >
+                <CheckIcon
+                  aria-label={
+                    value === "email"
+                      ? intl.formatMessage(commonMessages.email)
+                      : intl.formatMessage(commonMessages.inApp)
+                  }
+                  aria-hidden="false"
+                  data-h2-height="base(x1)"
+                  data-h2-width="base(x1)"
+                  data-h2-border="base(thin solid black.light)"
+                  data-h2-radius="base(input)"
+                  data-h2-vertical-align="base(middle)"
+                />
+              </span>
+            );
+          })}
+        </div>
+      </div>
+    );
+  }
```

Comment:
I am having a lot of problems trying to test the disabled buttons (especially with an AT enabled)

1. They are completely ignored by default so I can't even get to them
2. If I force by way onto them they are just announced as "email" and "inapp" with no context as to what that refers to which is kinda odd, what about email and in-app?
3. They are announced as images with just adds even more confusion
4. Visually, they don't look disabled to me, it just looks like a bug in the background color of the checkbox
5. Also, the icon weight does not match that of the actual checked boxes

Reply:
Ya the icons aren't part of the tab index so that can make it difficult to focus. Would adding the icons to the tab order work in this case? 

I can add aria messages to improve the announced message (e.g. System message email notifications are always on).

As for the other problems you mentioned I'm not sure. For the styling I could create a custom icon that matches the checkbox styling, but that's out of scope for this issue.

Reply:
> Ya the icons aren't part of the tab index so that can make it difficult to focus. Would adding the icons to the tab order work in this case, without effecting accessibility?

It would make them focusable but the problems with them being disabled are still there for both visual and non-visual user :confounded: The problem is, it states it is an image which is confusing and then it also does't really tell me much :woman_shrugging: 

> I can add aria messages to improve the announced message (e.g. System message email notifications are always on).

You could, but is this just not repeating the text to the left of these check boxes? It seems redundant imo.